### PR TITLE
Make sure to generate events when we clear credentials.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
We could end up in a situation where we cleared some credentials, then hit an error, and never generated the event for the credentials we had cleared.

### CODE REVIEW

File: src/app/clusters/door-lock-server/door-lock-server.cpp [hunk 1]
```diff
@@ -2899,23 +2899,32 @@ EmberAfStatus DoorLockServer::clearCredentials(chip::EndpointId endpointId, chip
...
-        auto status = clearCredential(endpointId, modifier, sourceNodeId, credentialType, i, false);
+        status = clearCredential(endpointId, modifier, sourceNodeId, credentialType, i, false);
         if (EMBER_ZCL_STATUS_SUCCESS != status)
         {
             ChipLogError(Zcl,
                          "[clearCredentials] Unable to clear the credential - internal error "
                          "[endpointId=%d,credentialType=%u,credentialIndex=%d,status=%d]",
                          endpointId, to_underlying(credentialType), i, status);
-            return status;
+            goto exit;
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:2913]: That's a good question.  Probably yes....

[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/clusters/door-lock-server/door-lock-server.cpp]: @jrhees-cae Done.

File: src/app/clusters/door-lock-server/door-lock-server.cpp [hunk 2]
```diff
@@ -2899,23 +2899,37 @@ EmberAfStatus DoorLockServer::clearCredentials(chip::EndpointId endpointId, chip
         return EMBER_ZCL_STATUS_FAILURE;
     }
 
+    EmberAfStatus status   = EMBER_ZCL_STATUS_SUCCESS;
+    bool clearedCredential = false;
     for (uint16_t i = 1; i < maxNumberOfCredentials; ++i)
     {
-        auto status = clearCredential(endpointId, modifier, sourceNodeId, credentialType, i, false);
-        if (EMBER_ZCL_STATUS_SUCCESS != status)
+        EmberAfStatus clearStatus = clearCredential(endpointId, modifier, sourceNodeId, credentialType, i, false);
```
Comment (andy31415) [inline src/app/clusters/door-lock-server/door-lock-server.cpp:2906]: that `false` should at least have a comment saying `/* sendUserChangeEvent */` or similar. Or better yet not be a bool and be `UserSendEvent::kDoNotSend` or similar.

Not part of this PR, just complaining that reviewing code is harder when such bools are used as arguments.

**Review by andy31415 — APPROVED** [review]

File: src/app/clusters/door-lock-server/door-lock-server.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:2906]: Yes, there's a lot of cleanup we could try to do here...  :(

**Review by woody-apple — APPROVED** [review]
